mslice/models/workspacemanager/file_io.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import (absolute_import, division, print_function)
+import os.path
+from mantid.api import IMDHistoWorkspace
+from mantid.simpleapi import CreateMDHistoWorkspace, SaveMD, SaveNexus, SaveAscii
+from mslice.util.qt.QtWidgets import QFileDialog
+
+import numpy as np
+from scipy.io import savemat
+
+
+def get_save_directory(multiple_files=False, save_as_image=False, default_ext=None):
+    '''
+    Show file dialog so user can choose where to save.
+    :param multiple_files: boolean - whether more than one file is being saved
+    :param save_as_image: boolean - whether to allow saving as a .png/pdf
+    :param default_ext: file extension that is selected by default
+    :return: path to save directory, name to save the file as, file format extension
+    :raises: RuntimeError if dialog is cancelled
+    '''
+    if multiple_files:
+        return QFileDialog.getExistingDirectory(), None, default_ext
+    else:
+        file_dialog = QFileDialog()
+        file_dialog.setAcceptMode(QFileDialog.AcceptSave)
+
+        filter = "Nexus (*.nxs);; Ascii (*.txt);; Matlab (*.mat)"
+        if save_as_image:
+            filter += ";; Image (*.png);; PDF (*.pdf)"
+        file_dialog.setNameFilter(filter)
+
+        if default_ext:
+            ext_to_qtfilter = {'.nxs': 'Nexus (*.nxs)', '.txt': 'Ascii (*.txt)', '.mat': 'Matlab (*.mat)', }
+            file_dialog.selectNameFilter(ext_to_qtfilter[default_ext])
+        if (file_dialog.exec_()):
+            path = file_dialog.selectedFiles()[0] + file_dialog.selectedFilter()[-5:-1]
+            ext = path[path.rfind('.'):]
+            return os.path.dirname(path), os.path.basename(path), ext
+        else:
+            raise RuntimeError("dialog cancelled")
+
+
+def save_nexus(workspace, path, is_slice):
+    if is_slice:
+        SaveMD(InputWorkspace=workspace.name()[2:], Filename=path)
+    elif isinstance(workspace, IMDHistoWorkspace):
+        SaveMD(InputWorkspace=workspace.name(), Filename=path)
+    else:
+        SaveNexus(InputWorkspace=workspace.name(), Filename=path)
+
+
+def save_ascii(workspace, path, is_slice):
+    if is_slice:
+        _save_slice_to_ascii(workspace, path)
+    elif isinstance(workspace, IMDHistoWorkspace):
+        _save_cut_to_ascii(workspace, workspace.name(), path)
+    else:
+        SaveAscii(InputWorkspace=workspace, Filename=path)
+
+
+def save_matlab(workspace, path, is_slice):
+    if is_slice:
+        x, y, e = _get_slice_mdhisto_xye(workspace)
```

Comment:
Could we leave the Matlab output as a 2D matrix rather than a collapsed vector? E.g. break `_get_slice_mdhisto_xye` at line 134 (after `e = `) and return the `x`, `y`, `e` values at that point for the matlab write-out, whilst leaving the rest (the `reshape`) for the `ascii` write-out?

This is because if I want to plot a 2D colorplot in Matlab with the current loaded data, I need to `reshape` it again into 2D form, which is a bit of a pain.